_docs/timeline/2017-01-02-bootstrap.md

```diff
@@ -22,3 +22,40 @@ The Bootstrap era will lead to the [Reward era](/timeline/reward), during which
 updates to the protocol will be issued, and the major stakeholders will be
 provided with convenient options to run their nodes on personal servers in the
 cloud.
+
+## Stake Locking
+
+Bootstrap era is the period of Cardano SL existence that allows only fixed predefined
+users to have control over the system. The set of such users (bootstrap stakeholders)
+is defined in [`gcdBootstrapStakeholders`](https://github.com/input-output-hk/cardano-sl/blob/f73d41b2bbd0a823911490974a71608fb4dbe014/core/Pos/Core/Genesis/Types.hs#L74). Bootstrap era will end when bootstrap stakeholders
+will vote for it. Next era after Bootstrap is called Reward era, as mentined [here](/timeline/reward/).
+
+Every transaction output (which is a pair `(address,coin)`) is accopmanied by transaction
+output distribution (`txOutDistr`). [`TxOutDistribution`](https://github.com/input-output-hk/cardano-sl/blob/f73d41b2bbd0a823911490974a71608fb4dbe014/txp/Pos/Txp/Core/Types.hs#L132) type is `[(StakeholderId, Coin)]` and it defines
+stake distribution of the output. Unlike balance (real amount of coins on the balance),
+stake gives user power to control different algorithm parts: being slot leader, voting
+in update system, taking part in MPC/SSC. `txOutDistr` is an ad-hoc way to delegate the stake
+without using more elaborate and complex delegation scheme. If `[]` is a set in `txOutDistr`,
+it's supposed to be `[(address, coin)]`, so stake goes to where output tells. It's important
+to notice that the sum of coins in `txOutDistr` should be equal to `coin`.
+
+Let's now present the Bootstrap era solution:
+
+1.  Initial `utxo` contains all the stake distributed among `gcdBootstrapStakeholders`. Initial `utxo`
+    consists of `(txOut, txOutDistr)` pairs, so we just set `txOutDistr` in a way it sends all coins
+    to `gcdBootstrapStakeholders`. Let `txOut = (address, coin)`. The distribution technique inside
+    attributes `coin / (length gcdBootstrapStakeholders)` to every party in `gcdBootstrapStakeholders`,
+    remainder to arbitrary `b ∈ gcdBootstrapStakeholders` (based on `hash coin`).
+2.  While Bootstrap era takes place users can send transactions changing initial `utxo`. We enforce
+    spreading `txDistr` to `gcdBootstrapStakeholders` in our wallet/client and forbid transactions that
+    move stake out of `gcdBootstrapStakeholders` set. This effectively makes stake distribution is system
+    constant.
+3.  When Bootstrap era is over, we disable transaction restrictions. System operates as it did in
+    Bootstrap era, but users need to explicitly state they understand owning their stake leads to
+    responsibility to handle the node. For user to get his stake back he should send a transaction
+    to himself.
```

Comment:
~to himself~ to delegate key(s). It may be key, owned by user himself or key of some delegate (which may also be one or few of `gcdBootstrapStakeholders`).